components/DataCard.tsx

```diff
@@ -1,37 +1,273 @@
 import React from "react";
 import {useState, useEffect} from "react";
+import Image from "next/image";
+import useSWR from "swr";
+import Fetcher from "../fetcher";
 
 
 interface DataCardProps {
     title: string;
-    data: string;
-    data1d: string[];
-    data7d: string[];
-    data30d: string[];
+    dataField: string;
 
 }
 
+function DataCard({ title, dataField }: DataCardProps) {
 
+    const [timeLine, setTimeLine] = useState<number>(3);
+    const [trend, setTrend] = useState<number>(0);
+    const [trendClass, setTrendClass] = useState<string>("body_trend_positive");
 
 
-function DataCard({ title, data, data1d, data7d, data30d}: DataCardProps) {
 
 
-    const [timeLine, setTimeLine] = useState<number>(100);
 
+    let dataArray: string[] = [];
 
+    const retrieveData = () => {
+        let gotchiverseStats7dResponse = useSWR(
+            "/api/gotchiverse/stats/7",
+            Fetcher
+        );
 
-    if (!data ) {
+        let gotchiverseStats1dResponse = useSWR(
```

Comment:
try to lazy load: fetch initial data of 1d an on click load data for 7 days and 30 days.